libraries/AP_BattMonitor/AP_BattMonitor.cpp

```diff
@@ -255,11 +255,15 @@ AP_BattMonitor::init()
 #ifdef HAL_BATT_MONITOR_DEFAULT
     _params[0]._type.set_default(int8_t(HAL_BATT_MONITOR_DEFAULT));
 #endif
+#ifdef HAL_BATT2_MONITOR_DEFAULT
+    _params[1]._type.set_default(int8_t(HAL_BATT2_MONITOR_DEFAULT));
```

Comment:
i think this may crash MatekL431-Periph, as it has this define and only has one battmon, so would index past end of the array

Reply:
could remove the define from MatekL431

Reply:
Did both.